tests/cpp/test_swizzle.cpp

```diff
@@ -690,4 +695,56 @@ TEST_F(SwizzleTest, SwizzleInProducerProjection) {
   testValidate(fusion.get(), outputs, {t}, {expect}, __LINE__, __FILE__);
 }
 
+TEST_F(SwizzleTest, Transpose1) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  auto tv0 = makeSymbolicTensor(2);
+  fusion.addInput(tv0);
+  auto tv1 = set(tv0);
+  auto tv2 = transpose(tv1, 0, 1);
+  fusion.addOutput(tv2);
+  tv1->setMemoryType(MemoryType::Shared);
+
+  std::vector<IterDomain*> dim0{tv1->axis(0), tv2->axis(1)};
+  std::vector<IterDomain*> dim1{tv1->axis(1), tv2->axis(0)};
+
+  AbstractTensor loop{dim0, dim1};
+  loop.split(1, 32);
+  loop.split(0, 32);
+  loop.reorder({{1, 2}});
+  loop.merge(0);
+  // BIDx, 32, 32
+
+  auto smem_alloc = loop.unzip()[0];
+  smem_alloc.swizzle(SwizzleType::XOR, 1, 2);
+
+  std::swap(loop[1][1], loop[2][1]);
+  loop.merge(1);
+  loop.split(1, 256);
+  // BIDx, 4, TIDx
+
+  for (auto id : loop[0].as<std::vector>()) {
+    id->parallelize(ParallelType::BIDx);
+  }
+  for (auto id : loop[2].as<std::vector>()) {
+    id->parallelize(ParallelType::TIDx);
+  }
+
+  auto ub = loop.unzip();
+  tv1->setLoopDomain(ub[0].as<IterDomain*>());
+  tv2->setLoopDomain(ub[1].as<IterDomain*>());
+  tv1->setAllocationDomain(smem_alloc.as<IterDomain*>(), true);
```

Comment:
Is this the reason of removing the check in `IterDomain::parallelize`? If `setLoopDomain` is done before `id->parallelize`, can we keep the check there?

Reply:
For this specific PR, yes, we can keep. But I don't think this check makes sense any more. Loop domain are no longer required to be leaf, so it makes no sense to assert that it has no use.

Reply:
Specifically, if don't parallelize the loop domain as `[4, TIDx{32*32/4}]`, but instead, as `[TIDy{32}, TIDx{32}]`, then this check will fail, even if we set loop domain first.

Reply:
I'd prefer keeping the check as long as we can. When the check was added, we did find a couple of inconsistent scheduling. Am I understanding correctly that we can still keep it for now?

Reply:
I'd be happy to change my mind then!